modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenConstants.java

```diff
@@ -359,4 +359,18 @@
     public static final String USE_SINGLE_REQUEST_PARAMETER = "useSingleRequestParameter";
     public static final String USE_SINGLE_REQUEST_PARAMETER_DESC = "Setting this property to true will generate functions with a single argument containing all API endpoint parameters instead of one argument per parameter.";
 
+    // The reason this parameter exists is because there is a dependency
+    // on swagger-api/swagger-parser issue https://github.com/swagger-api/swagger-parser/issues/1369.
+    // When that issue is resolved, this parameter should be removed.
+    public static final String LEGACY_ADDITIONAL_PROPERTIES_BEHAVIOR = "legacyAdditionalPropertiesBehavior";
+    public static final String LEGACY_ADDITIONAL_PROPERTIES_BEHAVIOR_DESC =
+        "If true, the 'additionalProperties' keyword is implemented as specified in the OAS and JSON schema specifications. " +
+        "Full compliance currently works with OAS 3.0 documents only. " +
+        "It is not supported for 2.0 documents until issues #1369 and #1371 have been resolved in the dependent swagger-parser project. " +
+        "If false, codegen uses a legacy, non-compliant interpretation of the 'additionalProperties' keyword. " +
```

Comment:
This description is confusing to me. We are describing it as legacyAdditionalPropertiesBehavior.
So it sounds like True mean s use legacy behavior.
But then in our definition we say true is spec compliant behavior
False is legacy non-compliant behavior here.
How about instead naming this something like:
- NULLABLE_ADDITIONAL_PROPERTIES_BEHAVIOR
- COMPLIANT_ADDITIONAL_PROPERTIES_BEHAVIOR
or keeping it:
- LEGACY_ADDITIONAL_PROPERTIES_BEHAVIOR
but have True be non-compliant legacy behavior, and False be compliant behavior

Also, what generator sets this to True?
I don't see that happening outside fo the tests. Did I miss it?

Reply:
Good catch, I inadvertently swapped the values.

Reply:
Improved documentation. I don't think it's necessary to change the name based on the improved documentation.

Reply:
The default value is set at https://github.com/OpenAPITools/openapi-generator/pull/6290/files#diff-7cb46fa53f89a458a7b7cb201d2214a8R1511